flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamingJobGraphGenerator.java

```diff
@@ -1178,6 +1180,27 @@ && isChainable(upStreamVertex.getInEdges().get(0), streamGraph)) {
         return upStreamVertex.getOperatorFactory();
     }
 
+    private void markSourcesAndSinks() {
```

Comment:
Maybe `markContainsSourcesAndSinks`

Reply:
Ok. Will rename it to be `markContainsSourcesOrSinks`.